### PR TITLE
Fix issue with current state callback

### PR DESCRIPTION
**Pull request by stathismor**
1. `Game` was registering a callback **before** the server emitted the event.
2. Fix issue with game state always being randomly generated.
3. Server now broadcasts the current state to all clients.

### CODE REVIEW

File: packages/server/src/rooms.ts [hunk 1]
```diff
@@ -55,12 +55,12 @@ async function doJoinRoom(socket: Socket, store: Store, room: Room, player: Play
   if (room.players.length === MAX_ROOM_SIZE) {
     let state = store.getCurrentState(room.name);
     if (!state) {
-      const playerNames = room.players.map((player) => player.name);
+      const playerNames = room.players.map((player) => playerName);
```
Comment (dbertella) [inline packages/server/src/rooms.ts]: This doesn't make sense it will return twice the same username

Comment (stathismor) [reply packages/server/src/rooms.ts]: Correct, done.

Comment (dbertella) [reply packages/server/src/rooms.ts]: Still doesn't :-) what's the logic? You are mapping over a list and returning a list with the same username? Why. What are you trying to do? I really don't get it

Comment (dbertella) [reply packages/server/src/rooms.ts]: You can remove the map and return [playername, playername] directly. Plus player the parameter of the function is not used here we need some eslint in the server too. are you sure it's what you want?

Comment (dbertella) [reply packages/server/src/rooms.ts]: Ok I see you removed but for some reason I didn't see the changes. Nevermind

**Review by dbertella — APPROVED** [review]

**Pull request merged**